src/core/queryClient.ts

```diff
@@ -132,15 +133,22 @@ export class QueryClient {
     })
   }
 
-  resetQueries(filters?: QueryFilters): void
-  resetQueries(queryKey?: QueryKey, filters?: QueryFilters): void
-  resetQueries(arg1?: QueryKey | QueryFilters, arg2?: QueryFilters): void {
-    const [filters] = parseFilterArgs(arg1, arg2)
+  resetQueries(filters?: QueryFilters, options?: ResetOptions): Promise<void>
+  resetQueries(queryKey?: QueryKey, filters?: QueryFilters, options?: ResetOptions): Promise<void>
+  resetQueries(arg1?: QueryKey | QueryFilters, arg2?: QueryFilters | ResetOptions, arg3?: ResetOptions): Promise<void> {
+    const [filters, options] = parseFilterArgs(arg1, arg2, arg3)
     const queryCache = this.queryCache
-    notifyManager.batch(() => {
+
+    const refetchFilters: QueryFilters = {
+      ...filters,
+      active: true,
+    }
+
+    return notifyManager.batch(() => {
       queryCache.findAll(filters).forEach(query => {
         query.reset()
       })
+      return this.refetchQueries(refetchFilters, options)
```

Comment:
Would it be better to have `query.reset()` trigger the fetch if the query active? That seems like better encapsulation, but it seems like the client coordinating things is more typical than the query doing it in this project, so far.